homeassistant/components/cookidoo/config_flow.py

```diff
@@ -0,0 +1,211 @@
+                suggested_values=self.reconfigure_entry.data,
+            ),
+            description_placeholders={CONF_NAME: self.reconfigure_entry.title},
+            errors=errors,
+        )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauth upon an API authentication error."""
+        await self.generate_schemata()
+        self.reauth_entry = self._get_reauth_entry()
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Dialog that informs the user that reauth is required."""
+        errors: dict[str, str] = {}
+
+        if TYPE_CHECKING:
+            assert self.reauth_entry
+
+        if user_input is not None:
+            if not (
+                errors := await self.validate_input(
+                    {**self.reauth_entry.data, **user_input}
+                )
+            ):
+                return self.async_update_reload_and_abort(
+                    self.reauth_entry, data_updates=user_input
+                )
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=self.add_suggested_values_to_schema(
+                data_schema=vol.Schema(AUTH_DATA_SCHEMA),
+                suggested_values={CONF_EMAIL: self.reauth_entry.data[CONF_EMAIL]},
+            ),
+            description_placeholders={CONF_NAME: self.reauth_entry.title},
+            errors=errors,
+        )
+
+    async def generate_schemata(self) -> None:
+        """Generate schemata."""
+        self.localizations = await get_localization_options()
+
+        self.LOCALIZATION_DATA_SCHEMA = {
+            vol.Required(CONF_LOCALIZATION): SelectSelector(
+                SelectSelectorConfig(
+                    mode=SelectSelectorMode.DROPDOWN,
+                    translation_key="localization",
+                    options=[
+                        LOCALIZATION_SPLIT_CHAR.join(
+                            [localization["country_code"], localization["language"]]
+                        ).lower()
+                        for localization in self.localizations
+                    ],
+                    sort=True,
+                ),
+            ),
+        }
+
+    async def validate_input(self, user_input: Mapping[str, Any]) -> dict[str, str]:
+        """Input Helper."""
+
+        errors: dict[str, str] = {}
+
+        session = async_get_clientsession(self.hass)
+        cookidoo = Cookidoo(
+            session,
+            {
+                **DEFAULT_COOKIDOO_CONFIG,
+                "localization": cookidoo_localization_for_key(
+                    self.localizations, user_input[CONF_LOCALIZATION]
+                ),
+                "email": user_input[CONF_EMAIL],
+                "password": user_input[CONF_PASSWORD],
```

Comment:
```suggestion
              **user_input,
               CONF_LOCALIZATION: cookidoo_localization_for_key(
                    self.localizations, user_input[CONF_LOCALIZATION]
                ),
```
I believe should do the same

Reply:
@zweckj Yes it does, bot the types are not happy